plugins/samples/check_pii/plugin.cc

```diff
@@ -22,64 +22,83 @@ class MyRootContext : public RootContext {
       : RootContext(id, root_id) {}
 
   bool onConfigure(size_t) override {
-    // Credit card numbers in a 19 character hyphenated format.
+    // Credit card numbers in a 16 character hyphenated format.
     // Compile the regex expression at plugin setup time, so that this expensive
     // operation is only performed once, and not repeated with each request.
     card_match.emplace("\\d{4}-\\d{4}-\\d{4}-(\\d{4})");
-    return card_match->ok();
+
+    // Compile the regex for 10-digit numeric codes.
+    code10_match.emplace("\\d{7}(\\d{3})");
+
+    // Ensure both regex patterns compiled successfully.
+    return card_match->ok() && code10_match->ok();
   }
 
   std::optional<re2::RE2> card_match;
+  std::optional<re2::RE2> code10_match;
 };
 
-// Checks the response http headers, and the response body for the presence of
-// credit card numbers. Mask the initial numbers case found.
+// Checks the response HTTP headers and the response body for the presence of
+// credit card numbers and 10-digit numeric codes. Masks the initial numbers
+// found while preserving the last few digits for both types of PII.
+//
+// - Credit Card Numbers: Masks the first 12 digits, displaying only the last 4 digits
+//   in the format XXXX-XXXX-XXXX-1234.
+// - 10-Digit Codes: Masks the first 7 digits, displaying only the last 3 digits
+//   as XXXXXXX123.
+//
 // Note that for illustrative purposes, this example is kept simple and does not
-// handle the case of credit card numbers that are split across multiple
-// onResponseBody() calls.
+// handle cases where PII is split across multiple onResponseBody() calls.
 class MyHttpContext : public Context {
  public:
   explicit MyHttpContext(uint32_t id, RootContext* root)
       : Context(id, root), root_(static_cast<MyRootContext*>(root)) {}
 
   FilterHeadersStatus onResponseHeaders(uint32_t headers,
                                         bool end_of_stream) override {
-    const auto pii_header = getResponseHeader("google-run-pii-check");
-    if (pii_header && pii_header->view() == "true") {
-      check_body_ = true;
-      const auto result = getResponseHeaderPairs();
-      const auto pairs = result->pairs();
-      for (auto& p : pairs) {
-        std::string header_value = std::string(p.second);  // mutable copy
-        if (maskCardNumbers(header_value)) {
-          replaceResponseHeader(p.first, header_value);
-        }
+    const auto result = getResponseHeaderPairs();
+    const auto pairs = result->pairs();
+    for (auto& p : pairs) {
+      std::string header_value = std::string(p.second);  // mutable copy
+      if (maskPII(header_value)) {
+        replaceResponseHeader(p.first, header_value);
```

Comment:
This is a bit of an uncommon case, but if a header appears multiple times in the request, this logic may end up collapsing the multiple occurrences into a single entry. For example, suppose the request has both of the following headers:

"credit-card: 0123-4567-8901-234"
"credit-card: 5678-9012-3456-7890"

Then after this processing, the response would only have the single header "credit-card: XXXX-XXXX-XXXX-7890".

You can avoid this problem by instead collecting the modified values in a HeaderStringPairs vector, and setting them all at once, e.g.:

```
const HeaderStringPairs orig_headers = getResponseHeaderPairs();
HeaderStringPairs new_headers;
bool changed = false;
for (auto [key, value] : orig_headers) {
  changed = changed || maskPII(value);
  new_headers.emplace_back(std::make_pair(std::move(key), std::move(value));
}
if (changed) {
  setResponseHeaderPairs(new_headers);
}
```

Reply:
I tried using
```c++
const HeaderStringPairs orig_headers = getResponseHeaderPairs();
``` 
but I was getting
```
error: no viable conversion from 'WasmDataPtr' (aka 'unique_ptr<WasmData>') to 'const HeaderStringPairs' (aka 'const vector<pair<basic_string<char>, basic_string<char>>>')
``` 
so I tried a workaround using the original approach, I believe it should work as well, let me know what you think